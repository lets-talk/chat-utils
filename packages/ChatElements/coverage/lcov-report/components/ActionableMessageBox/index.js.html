<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/ActionableMessageBox/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/ActionableMessageBox</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
&nbsp;
import './index.scss';
&nbsp;
const ActionableMessageBox = (props) =&gt; {
  const { data, onClickAction } = props;
  const { actions } = data;
&nbsp;
  return (
    &lt;ul className={classNames('letstalk-actionable-mbox')}&gt;
      {actions.map((action, index) =&gt;
        (
          &lt;li
            key={action.id}
            className="letstalk-actionable-item"
            role="button"
            onClick={onClickAction &amp;&amp; (<span class="fstat-no" title="function not covered" ><span class="branch-1 cbranch-no" title="branch not covered" >(e</span>) =&gt; <span class="cstat-no" title="statement not covered" >onClickAction(action, index, e))</span></span>}
          &gt;
            {action.name}
          &lt;/li&gt;
        ))}
    &lt;/ul&gt;
  );
};
&nbsp;
ActionableMessageBox.defaultProps = {
  onClickAction: null,
};
&nbsp;
ActionableMessageBox.propTypes = {
  data: PropTypes.object.isRequired,
  onClickAction: PropTypes.func,
};
&nbsp;
export default ActionableMessageBox;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 24 2017 01:32:25 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
