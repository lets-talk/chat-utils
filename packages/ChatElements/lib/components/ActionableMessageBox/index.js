import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import FaCloudDownload from 'react-icons/lib/fa/cloud-download';

import './index.scss';

const ActionableMessageBox = (props) => {
  const { data, onClickAction } = props;
  const { actions } = data;

  return (
    <ul className={classNames('letstalk-actionable-mbox')}>
      {actions.map((action, index) =>
        (
          <li
            key={action.id}
            className="letstalk-actionable-item"
            role="button"
            onClick={onClickAction && ((e) => onClickAction(action, index, e))}
          >
            {action.name}
          </li>
        ))}
    </ul>
  );
};

ActionableMessageBox.defaultProps = {
  data: {},
  onClickAction: null,
};

ActionableMessageBox.propTypes = {
  data: PropTypes.object,
  onClickAction: PropTypes.func,
};

export default ActionableMessageBox;
