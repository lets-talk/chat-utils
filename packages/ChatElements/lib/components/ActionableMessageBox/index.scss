@import './config.scss';

.letstalk-actionable-mbox {
  display: block;
  cursor: pointer;
  padding: 0;
  margin: 0;
  border-radius: 5px;
  background-color: $color-white;
  font-size: $font-size-caption;
  font-weight: $font-weight-semi-bold;

  li.letstalk-actionable-item {
    list-style: none;
    padding: 15px;
    @include themify($themes) {
      color: themed('color-background-primary');
    }
  }

  li.letstalk-actionable-item:hover {
    @include themify($themes) {
      background-color: rgba(themed('color-background-primary'), .05);
    }
  }

  li.letstalk-actionable-item:first-child {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
  }

  li.letstalk-actionable-item:last-child {
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
  }

  li.letstalk-actionable-item:not(:last-child) {
    @include themify($themes) {
      border-bottom: 1px solid themed('color-divider');
    }
  }

}
