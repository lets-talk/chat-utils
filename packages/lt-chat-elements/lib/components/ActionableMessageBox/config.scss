@import "../../common/styles/variables";
@import "../../common/styles/themes/themify";

$input-border-color: rgb(136, 196, 63);
