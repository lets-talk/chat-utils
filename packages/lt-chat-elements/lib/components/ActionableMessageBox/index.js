import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './index.scss';

const ActionableMessageBox = (props) => {
  const { data, onClickAction } = props;
  const { actions } = data;

  return (
    <ul className={classNames('letstalk-actionable-mbox')}>
      {actions.map((action, index) =>
        (
          <li
            key={action.id}
            className="letstalk-actionable-item"
            role="button"
            onClick={onClickAction && ((e) => onClickAction(action, index, e))}
          >
            <span>{action.name}</span>
          </li>
        ))}
    </ul>
  );
};

ActionableMessageBox.defaultProps = {
  onClickAction: null,
};

ActionableMessageBox.propTypes = {
  data: PropTypes.object.isRequired,
  onClickAction: PropTypes.func,
};

export default ActionableMessageBox;
