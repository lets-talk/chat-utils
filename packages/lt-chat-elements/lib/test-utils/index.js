// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
