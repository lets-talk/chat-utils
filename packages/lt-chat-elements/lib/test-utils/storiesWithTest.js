import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
