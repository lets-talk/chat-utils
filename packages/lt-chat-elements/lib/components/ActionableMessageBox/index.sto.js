import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, select, text } from '@storybook/addon-knobs';

// Component to show on storybook
import ActionableMessageBox from '.';
import { WrapWithTheme } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';

const stories = storiesOf('ActionableMessageBox', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('ActionableMessageBox'));

const data = {
  actions: [
    {
      id: 1,
      name: 'Emergencia Bancaria',
    },
    {
      id: 2,
      name: 'Tarjeta de crédito',
    },
    {
      id: 3,
      name: 'Crédito Personal',
    },
    {
      id: 4,
      name: 'Otro',
    },
  ],
};

stories.addWithInfo(
  'ActionableMessageBox',
  'This is the basic usage with the ActionableMessageBox',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <WrapWithTheme
        themeName={select('Theme', { default: 'DefaultTheme', light: 'LightTheme', dark: 'DarkTheme' }, 'default')}
      >
        <ActionableMessageBox
          onClickAction={action('onClick')}
          data={data}
        />
      </WrapWithTheme>
    </div>
  )
);
