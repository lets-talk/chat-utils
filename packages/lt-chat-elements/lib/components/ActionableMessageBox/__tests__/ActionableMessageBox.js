import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import ActionableMessageBox from '../index';
import { renderWithTheme, staticWithTheme, shallowWithTheme } from '../../../test-utils';

describe('ActionableMessageBox component', () => {
  const data = {
    actions: [
      {
        id: 1,
        name: 'Emergencia Bancaria',
      },
      {
        id: 2,
        name: 'Tarjeta de crédito',
      },
      {
        id: 3,
        name: 'Crédito Personal',
      },
      {
        id: 4,
        name: 'Otro',
      },
    ],
  };

  it('should render without data', () => {
    const component = shallow(<ActionableMessageBox data={data} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data', () => {
    const component = shallow(<ActionableMessageBox data={data} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have the correct styles (Those that depends on the theme)', () => {
    const tree = renderWithTheme(<ActionableMessageBox data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background-color', 'rgb(255,255,255)');
    expect(tree).toHaveStyleRule('font-size', 'calc(12 * 1px)');
    expect(tree).toHaveStyleRule('line-height', 'calc(16 * 1px)');
    expect(tree).toHaveStyleRule('font-weight', '500');
  });

  it('should render the correct html structure', () => {
    const wrapper = staticWithTheme(<ActionableMessageBox data={data} />);
    expect(wrapper.find('li')).toHaveLength(4);
  });

  it('should handle the click event on the items correctly', () => {
    const onClickAction = jest.fn();
    const wrapper = shallowWithTheme(<ActionableMessageBox data={data} onClickAction={onClickAction} />);

    wrapper.find('li').last().simulate('click');
    // The function is called with the item clicked, the index and the event (item, index, e)
    expect(onClickAction).toHaveBeenCalledWith(data.actions[3], 3, undefined);

    onClickAction.mockClear();
    wrapper.find('li').first().simulate('click');
    // The function is called with the item clicked, the index and the event (item, index, e)
    expect(onClickAction).toHaveBeenCalledWith(data.actions[0], 0, undefined);
  });
});
