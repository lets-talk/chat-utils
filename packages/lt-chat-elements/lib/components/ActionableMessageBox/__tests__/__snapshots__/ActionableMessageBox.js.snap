// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionableMessageBox component should render with data 1`] = `
<ul
  className="letstalk-actionable-mbox"
>
  <li
    className="letstalk-actionable-item"
    key="1"
    onClick={null}
    role="button"
  >
    <span>
      Emergencia Bancaria
    </span>
  </li>
  <li
    className="letstalk-actionable-item"
    key="2"
    onClick={null}
    role="button"
  >
    <span>
      Tarjeta de crédito
    </span>
  </li>
  <li
    className="letstalk-actionable-item"
    key="3"
    onClick={null}
    role="button"
  >
    <span>
      Crédito Personal
    </span>
  </li>
  <li
    className="letstalk-actionable-item"
    key="4"
    onClick={null}
    role="button"
  >
    <span>
      Otro
    </span>
  </li>
</ul>
`;

exports[`ActionableMessageBox component should render without data 1`] = `
<ul
  className="letstalk-actionable-mbox"
>
  <li
    className="letstalk-actionable-item"
    key="1"
    onClick={null}
    role="button"
  >
    <span>
      Emergencia Bancaria
    </span>
  </li>
  <li
    className="letstalk-actionable-item"
    key="2"
    onClick={null}
    role="button"
  >
    <span>
      Tarjeta de crédito
    </span>
  </li>
  <li
    className="letstalk-actionable-item"
    key="3"
    onClick={null}
    role="button"
  >
    <span>
      Crédito Personal
    </span>
  </li>
  <li
    className="letstalk-actionable-item"
    key="4"
    onClick={null}
    role="button"
  >
    <span>
      Otro
    </span>
  </li>
</ul>
`;
